src/metadata.c

```diff
@@ -161,8 +153,7 @@ ts_metadata_insert(Datum metadata_key, Oid key_type, Datum metadata_value, Oid v
 	}
 
 	/* Insert into the catalog table for persistence */
-	values[AttrNumberGetAttrOffset(Anum_metadata_key)] =
-		convert_type_to_name(metadata_key, key_type);
+	values[AttrNumberGetAttrOffset(Anum_metadata_key)] = CStringGetDatum(metadata_key);
```

Comment:
This is going to crash with ASAN, since it's a `NameData` and `heap_form_tuple()` is going to `memcpy()` it without checking the length. You should have a `palloc0`ed name here.

Reply:
changed

Reply:
`  strlcpy(metadata_key, key_data, NAMEDATALEN);` are destination and source in the wrong order?

Reply:
You could use the structure:
```
typedef struct nameData
{
	char		data[NAMEDATALEN];
} NameData;
```